### PR TITLE
Created UI for creating, editing and viewing rooms

### PR DESCRIPTION
**Pull request by claynicholson**

### CODE REVIEW

**Review by Cosmin-Mare — APPROVED** [review]
lgtm! doesn't influence anything outisde the beta workflow. But there's no close button, and there should be a view password button, so u can continue working on that and I'm going to start implementing the actual functionality into the UI

**Review by JosiasAurel — CHANGES_REQUESTED** [review]

File: src/components/design-system/input.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ interface InputProps {
 	bind?: Signal<string>
 	onChange: (event: any) => void
 	value: string
+	readonly?: boolean
```
[the remark this answers is not in the document]
Comment (claynicholson) [reply src/components/design-system/input.tsx:14]: The box is meant to meant to display the link to get copied. I wanted to use the same style from the other input boxes so  it just acts as a display box.

File: src/components/popups-etc/share-room.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+					</div>
+
+					<div class={styles.inputRow}>
+					<Input onChange={() => undefined} 
+					value="www.example.com" 
+					readonly
+					/>
+					<Button accent type='submit' disabled={false}
+					onClick={() => {
+						const textToCopy = "www.example.com";
```
[the remark this answers is not in the document]
Comment (claynicholson) [reply src/components/popups-etc/share-room.tsx:88]: Yes it does, I believe Cosmin is adding the link generation to it so I am just using it as a placeholder.

File: src/components/popups-etc/share-room.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+					<div class={styles.info}>
+						<p>Room ID: {props.roomState?.value.roomId}</p>
+						<p>Participants: {props.roomState?.value.participants.length}</p>
+						<p>Room Password: {props.roomState?.value.password}</p>	
+					</div>
+
+					<div class={styles.inputRow}>
+					<Input onChange={() => undefined} 
+					value="www.example.com"
```
[the remark this answers is not in the document]
Comment (claynicholson) [reply src/components/popups-etc/share-room.tsx:83]: alright, will do in the future.

File: src/components/popups-etc/share-room.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+					/>
+					<Button accent type='submit' disabled={false}
+					onClick={() => {
+						const textToCopy = "www.example.com";
+   						 navigator.clipboard.writeText(textToCopy)
+    					  .then(() => {
+      						  console.log('Text copied to clipboard');
+     						 })
+     					 .catch(err => {
+       						 console.error('Failed to copy text: ', err);
```
[the remark this answers is not in the document]
Comment (claynicholson) [reply src/components/popups-etc/share-room.tsx:94]: Alright Ill add that

File: src/components/popups-etc/share-room.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+   						   });
+						}}>
+								Copy
+							</Button>
+					</div>
+
+					
+					<form onSubmit={async (event) => {
```
[the remark this answers is not in the document]
Comment (claynicholson) [reply src/components/popups-etc/share-room.tsx:102]: Cosmin is implementing this I believe.

**Comment by Cosmin-Mare** [comment]
@JosiasAurel I'll add the stuff u mentioned, I just need this code on the main branch so I can work on it. It won't influence users anyways for now, that's what I need confirmation for

File: src/components/popups-etc/share-room.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (grymmy) [reply src/components/popups-etc/share-room.tsx:83]: The future is now, plz make this change.

File: src/components/popups-etc/share-room.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (grymmy) [reply src/components/popups-etc/share-room.tsx:102]: Seeing as this doesn't effect prod, it is fine for this to be a placeholder.

**Review by Cosmin-Mare — APPROVED** [review]
lgtm! @JosiasAurel do we have your blessing?:))

**Review by JosiasAurel — APPROVED** [review]
blessed!

**Pull request merged**